src/components/UI/Inputs/CurrencyInput/index.tsx

```diff
@@ -0,0 +1,170 @@
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from 'styled-components';
+
+import { InputElement } from '../Input';
+
+const InputWrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.colors.darkBlueLighter1};
+  border-radius: ${({ theme }) => theme.border_radius};
+  color: ${(props) => props.theme.colors.whiteInput};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledInput = styled(InputElement)`
+  border-radius: ${({ theme }) => theme.border_radius};
+  border: 0;
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+`;
+
+const CurrencyLabel = styled.div`
+  color: ${(props) => props.theme.colors.whiteInput};
+  background: ${({ theme }) => theme.colors.darkBlueLighter1};
+  color: ${({ theme }) => theme.colors.white5};
+  padding: 8px 10px;
+  line-height: 1;
+`;
+
+const LabelText = styled.label`
+  font-size: 14px;
+  color: ${(props) => props.theme.colors.white2};
+`;
+
+interface ICurrencyInputProps
+  extends Omit<
+    React.ComponentPropsWithRef<'input'>,
+    'type' | 'onChange' | 'step'
+  > {
+  currencyLabel?: string;
+  precision?: number;
+  value?: number;
+  label?: string;
+  min?: number;
+  max?: number;
+  labelTextProps?: React.ComponentPropsWithRef<'label'>;
+  rootProps?: Omit<React.ComponentPropsWithRef<'div'>, 'htmlFor'>;
+  onChange?: (newValue: number) => void;
+}
+
+const CurrencyInput: React.FC<ICurrencyInputProps> = ({
+  currencyLabel,
+  precision,
+  label,
+  labelTextProps,
+  rootProps,
+  value,
+  id,
+  min,
+  max,
+  onChange,
+  ...rest
+}) => {
+  // eslint-disable-next-line no-magic-numbers
+  const step = 1 / 10 ** precision!;
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let valueConvertedToPrecision = formatNumberWithPrecision(
+      e.target.valueAsNumber,
+      precision!
+    );
+
+    if (min) {
+      valueConvertedToPrecision = Math.max(valueConvertedToPrecision, min);
+    }
+    if (max) {
+      valueConvertedToPrecision = Math.min(valueConvertedToPrecision, max);
+    }
+
+    onChange?.(valueConvertedToPrecision);
+  };
+
+  const visibleValue = formatNumberWithPrecision(value ?? 0, precision!);
+
+  return (
+    <div {...rootProps}>
+      {label && (
+        <LabelText {...labelTextProps} htmlFor={id}>
+          {label}
+        </LabelText>
+      )}
+      <InputWrapper>
+        <CurrencyLabel>{currencyLabel}</CurrencyLabel>
```

Comment:
Could we instead add this CurrencyLabel as a feature of the base input component? not sure what to call it, maybe  "InputIcon"

So that we don't have to unset the border of the Input here to only rewrap it again..

and so I can reuse it for search input component 😅